src/app/core/parsers/zod-entity-parser.service.ts

```diff
@@ -166,44 +197,21 @@ export class ZodEntityParser implements EntityParser {
     return this.showStyleVariantParser.parse(showStyleVariant)
   }
 
-  private readonly studioConfigurationParser = zod.object({
-    settings: zod.object({
-      mediaPreviewUrl: zod.string().min(9, 'Media preview url must be longer than 9 characters long i.e. http://tld'),
-    }),
-    blueprintConfiguration: zod.object({
-      ServerPostrollDuration: zod.number().min(0, 'Server post-roll duration must be 0 or more.'),
-    }),
-  })
-
   public parseStudioConfiguration(studioConfiguration: unknown): StudioConfiguration {
     return this.studioConfigurationParser.parse(studioConfiguration)
   }
 
-  private readonly tv2ActionParser = zod.object({
-    id: zod.string(),
-    name: zod.string(),
-    type: zod.nativeEnum(PartActionType).or(zod.nativeEnum(PieceActionType)),
-    metadata: zod.object({
-      contentType: zod.nativeEnum(Tv2ActionContentType),
-    }),
-  })
-
   public parseTv2Action(tv2Action: unknown): Tv2Action {
     return this.tv2ActionParser.parse(tv2Action)
   }
 
-  private readonly mediaDataParser = zod.object({
-    id: zod.string(),
-    duration: zod.number(),
-  })
-
-  public parseMedia(media: unknown): Media {
-    return this.mediaDataParser.parse(media)
+  public parseMediaAsset(media: unknown): Media {
+    return this.mediaAssetParser.parse(media)
   }
 
-  private readonly systemInformationParser = zod.object({
-    name: zod.string(),
-  })
+  public parseMediaAssets(media: unknown): Media[] {
+    return this.mediaAssetsParser.parse(media)
+  }
```

Comment:
Seems like one of these two methods are redundant

Reply:
They both are used, unless i don't understand what you are referencing. The plural `parseMediaAssets` is used when we gather all media assets from Sofie Server, and the `parseMediaAsset` for when we receive media through events such as created and deleted events.

Reply:
I guess I got confused about `media` is both singular and plural :p